fixpatches/src/commonMain/kotlin/com/saveourtool/sarifutils/adapter/SarifFixAdapter.kt

```diff
@@ -215,42 +227,89 @@ class SarifFixAdapter(
      * @param fileReplacementsList list of replacements from all rules
      * @param targetFiles list of target files
      */
-    private fun applyReplacementsToFiles(fileReplacementsList: List<FileReplacements>, targetFiles: List<Path>): List<Path> = fileReplacementsList.mapNotNull { fileReplacements ->
-        val targetFile = targetFiles.find {
-            val fullPathOfFileFromSarif = if (!fileReplacements.filePath.adaptedIsAbsolute()) {
-                fs.canonicalize(sarifFile.parent!! / fileReplacements.filePath)
+    @Suppress("MaxLineLength")
+    private fun applyReplacementsToFiles(
+        fileReplacementsList: List<FileReplacements>,
+        targetFiles: List<Path>,
+    ): List<Path> {
+        if (fileReplacementsList.isEmpty()) {
+            log.warn { "The list of replacements is empty." }
+        }
+        if (targetFiles.isEmpty()) {
+            log.warn { "The list of replacements is empty." }
+        }
+
+        return fileReplacementsList.mapNotNull { fileReplacements ->
+            val fileUri = fileReplacements.filePath
+            log.info { "Processing file at URI: $fileUri" }
+            val localPath = try {
+                Uri(fileUri.pathString).toLocalPathExt()
+            } catch (_: IllegalArgumentException) {
+                /*
+                 * `fileUri` is actually a path, most probably a Windows path.
+                 */
+                fileUri
+            }
+
+            val absolute = localPath.isAbsolute
+            if (localPath != fileUri) {
+                log.info { "Resolved the URI to a local path: (absolute = $absolute): $localPath" }
+            }
+
+            /*
+             * No need to check whether `localPath` is absolute: if it is,
+             * `resolve()` will ignore `sarifFile.parent` and return `localPath`
+             * intact.
+             */
+            val absoluteLocalPath = (sarifFile.parent!! / localPath).normalized()
+            if (absoluteLocalPath != localPath) {
+                log.info { "Converted the path: $localPath -> $absoluteLocalPath" }
```

Comment:
Maybe it's a debug logging?

Reply:
The log level of `save-cli` (`save -- log all`) doesn't seem to affect `sarif-utils`.

Let's discuss this later -- my current focus is making `save-cli` stable.

Reply:
Referenced from #40, marking as resolved for now.